keymanapp/keyman
# chore(mac): use core actions struct

Instead of processing an array of action structs, get a single struct from core that encapsulates all the impacts of the most recently processed key.
 
Fixes #7916
Fixes #7542 
Fixes #7526 
Fixes #7021
Fixes #5012
Fixes #10116

# User Testing

**TEST_AMHARIC_STICKIES**: 

- Open the Stickies app on the Mac and create a new note
- Switch to the gff_amharic keyboard 
- Type <kbd>s</kbd><kbd>e</kbd><kbd>l</kbd><kbd>a</kbd><kbd>m</kbd>
- Verify that the output is ሰላም

**TEST_AMHARIC_TEXSTUDIO_9506**: 

- Open texstudio on the Mac and create a new document
- Switch to the gff_amharic keyboard
- Type <kbd>s</kbd><kbd>e</kbd><kbd>l</kbd><kbd>a</kbd><kbd>m</kbd>
- Verify that the output is ሰላም

**TEST_YIDDISH_STICKIES**: 

- Open the Stickies app on the Mac and create a new note
- Switch to the Yiddish Pasekh keyboard
- Type <kbd>k</kbd><kbd>e</kbd><kbd>y</kbd>
- Confirm that ouput is קײ
- Type <kbd>r</kbd><kbd>e</kbd><kbd>y</kbd>
- Verify that ouput is רײ

**TEST_YIDDISH_GOOGLE_DOCS_7916**: 

- Open an empty document in Google Docs
- Switch to the Yiddish Pasekh keyboard
- Type <kbd>k</kbd><kbd>e</kbd><kbd>y</kbd>
- Confirm that ouput is קײ
- Type <kbd>r</kbd><kbd>e</kbd><kbd>y</kbd>
- Verify that ouput is רײ

**TEST_KENYA_BTL_STICKIES**: 

- Open the Stickies app on the Mac and create a new note
- Switch to the Kenya (BTL) keyboard
- Type <kbd>b</kbd><kbd>^</kbd><kbd>u</kbd><kbd>g</kbd>
- Verify that ouput is bûg

**TEST_KENYA_BTL_GOOGLE_CHROME_7542**: 

- Open Google Chrome on the Mac
- Switch to the Kenya (BTL) keyboard
- Click on the URL field and type <kbd>b</kbd><kbd>^</kbd><kbd>u</kbd><kbd>g</kbd>
- Verify that ouput is bûg

**TEST_HEBREW_ENTER_STICKIES**: 

- Open the Stickies app on the Mac and create a new note
- Switch to the Galaxie Hebrew (Positional) keyboard
- Type <kbd>m</kbd><kbd>m</kbd>
- Verify that ouput is ממ
- Type <kbd>return</kbd>
- Verify that ouput has changed to מם

**TEST_POLYTONIC_GREEK_CHROME_BACKSPACE_7526**: 
- Open Google Chrome and select the URL text field
- Switch to the keyboard 'Polytonic Greek (SIL)'
- Type <kbd>a</kbd><kbd>b</kbd><kbd>c</kbd><kbd> backspace </kbd><kbd> backspace </kbd><kbd> backspace </kbd>
- Verify that the output changes as follows: ἀ, ἀβ, ἀβχ, ἀβ, ἀ, [empty]

**TEST_POLYTONIC_GREEK_BACKSPACE_7021**: 
- Open the Stickies app and create a new note
- Switch to the keyboard 'Polytonic Greek (SIL)'
- Type <kbd>a</kbd><kbd>n</kbd><kbd>t</kbd><kbd>o</kbd><kbd>s</kbd>
- Verify that pressing backspace will delete one character at a time
- Five backspaces will delete the entire word
